### DISCUSSION TITLE
Annoying coilgun microstuns

### DISCUSSION THREAD

**Discussion by ClayTeck** [FakeFishGames/Barotrauma#9712] [Ideas and suggestions]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

Coilgun microstuns are really annoying and it looks like they are not needed at all, and i see no reason for this shit. They literally do nothing useful but they weird. A few updates ago there were no microstuns to the coilguns, and it was ok. Of course microstuns are good mechanic, but definitely not for default/piercing ammo. It's okay to stun crawlers with explosive or physicorium ammo, or microstunning spineling with SMG. But, it really begins to annoy every living thing. When dumb teammate trying to shoot spineling, shoots you, and you die. Not because he shot you, but because he microstunned you, and you were not able to heal yourself. Or while you try to escape raptor on scooter, and coilgun also stops you, so you become slower, and die. Please bring old coilguns without microstun back. There are really no any balance reason or something for this shit to exist. It's just annoys, interferes with playing, and does nothing else. Coilgun microstun sucks

### Reproduction steps

1. Start a game
2. Invite your friend
3. Ask your friend to shoot you with coilgun
4. Try to do something while recieving infinite microstuns
5. Get fucked up

### Bug prevalence

Happens every time I play

### Version

0.18.15.0

### Which operating system did you encounter this bug on?

Windows

**Comment by Ek-Videogames** [comment]
Attack stun is a longstanding design challenge.

- Same problem with dumping a whole magazine into somebody with SMG.

It is a minor annoyance for AI creatures/characters, as they act with perfect frame-priority,
but for players its absolutely debilitating and very lethal as it disables all inventory and item interaction while ragdolled.

**Comment by ClayTeck** [comment]
Attack stuns worked well, until recent "balance" updates. It's fine to stun someone with SGM, as long you can't spam it from very long range for a very long time. Also, SMG don't make nothing weird. It's useful and won't able to bother you to death. But coilguns have a higher chance of friendly-fire, so microstuns should exist, but shouldn't be "rebalanced". It's just worked well. Attack stuns is only necessary for short-range weapons like SMG or shotguns, or long-range weapons with high damage like chainguns, pulse lasers etc.

**Reply by Ek-Videogames** [reply]
I don't think I understand your point - both of these attacks are high fire-rate weapons which stun you every single hit (thus rapidly canceling all actions & character inventory), how are they different in regards to that problem?
